### ISSUE TITLE
branching model of pygit2

### ISSUE THREAD

**Issue by cholin** [libgit2/pygit2#162]
I think we should switch to a different branching model for pygit2. At the moment we only have two branches `master` and `gh-pages`. I think we should add `development` as a third branch and use it as our main branch for the upcoming release. With that the `master` could be in sync with libgit2's `master` and we would be able to fix bugs for the current release and as well implement new features of libgit2's development branch in `development`. I think this would as well compensate sporadic releases by the libgit2 team.

In short 3 branches:
- `master` compiles with master of libgit2
- `development` main branch for upcoming releases (in sync with development of libgit2)
- `gh-pages` api docs

What do you think?
